src/mongo/delegates/Views.class.php

```diff
@@ -599,8 +600,37 @@ protected function doJoins($source, $joins, &$dest, $from, $contextAlias, $build
                     {
                         // make sure any sequences are expanded before extracting properties
                         if (isset($ruleset['joins'])) $this->expandSequence($ruleset['joins'],$linkMatch);
-
-                        $dest[_GRAPHS][] = $this->extractProperties($linkMatch,$ruleset,$from);
+                        if (isset($ruleset['filter']))
+                        {
+                            foreach($ruleset['filter'] as $filterPredicate => $filter)
+                            {
+                                foreach($filter as $filterType => $filterMatch)
+                                {
+                                    if (isset($linkMatch[$filterPredicate]))
```

Comment:
I think we might need to check for either the alias or the fquri here.  The data could potentially be inconsistent.